src/libcore/task/spawn_error.rs

```diff
@@ -49,3 +50,33 @@ pub struct SpawnObjError {
     /// The task for which spawning was attempted
     pub task: TaskObj,
 }
+
+/// The result of a failed spawn
+#[derive(Debug)]
+pub struct SpawnLocalObjError {
+    /// The kind of error
+    pub kind: SpawnErrorKind,
+
+    /// The task for which spawning was attempted
+    pub task: LocalTaskObj,
+}
+
+impl SpawnLocalObjError {
+    /// Converts the `SpawnLocalObjError` into a `SpawnObjError`
+    /// To make this operation safe one has to ensure that the `UnsafeTask`
+    /// instance from which the `LocalTaskObj` stored inside was created
+    /// actually implements `Send`.
+    pub unsafe fn as_spawn_obj_error(self) -> SpawnObjError {
+        // Safety: Both structs have the same memory layout
+        mem::transmute::<SpawnLocalObjError, SpawnObjError>(self)
+    }
+}
+
+impl From<SpawnObjError> for SpawnLocalObjError {
+    fn from(error: SpawnObjError) -> SpawnLocalObjError {
+        unsafe {
+            // Safety: Both structs have the same memory layout
+            mem::transmute::<SpawnObjError, SpawnLocalObjError>(error)
```

Comment:
No need to `transmute` here-- just move the fields over.

Reply:
Will it have the same performance?

Reply:
Yup.